includes/api/class-wc-rest-shipping-zone-methods-controller.php

```diff
@@ -103,6 +126,195 @@ public function get_items( $request ) {
 	}
 
 	/**
+	 * Create a new shipping zone method instance.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Request|WP_Error
+	 */
+	public function create_item( $request ) {
+		global $wpdb;
+
+		$method_id = $request['method_id'];
+		$zone      = $this->get_zone( $request['zone_id'] );
+		if ( is_wp_error( $zone ) ) {
+			return $zone;
+		}
+
+		$instance_id     = 0;
+		$wc_shipping     = WC_Shipping::instance();
+		$allowed_classes = $wc_shipping->get_shipping_method_class_names();
+		$count           = $wpdb->get_var( $wpdb->prepare( "SELECT COUNT(*) FROM {$wpdb->prefix}woocommerce_shipping_zone_methods WHERE zone_id = %d", $zone->get_id() ) );
+		if ( in_array( $method_id, array_keys( $allowed_classes ) ) ) {
```

Comment:
Is there a reason zone's own `add_shipping_method()` method cannot be reused here?